### PR TITLE
adding a normalization method

### PR DESCRIPTION
**Pull request by amir-zeraati**
Adding normal_seq as a normalization function that saves the constants for future transformation on new data-set.
this technique can be useful for new data and additionally it makes it possible to normalize the test set on the scale of training set and avoiding data leakage.

### CODE REVIEW

File: src/repli1d/nn.py [hunk 1]
```diff
@@ -95,25 +143,31 @@ def load_signal(name,
     for col in df.columns:
```
Comment (organic-chemistry) [inline src/repli1d/nn.py]: I think it is better if you put your code: 
```python
  if transform_norm == normal_seq:
      df = pd.DataFrame(transform_norm(df))
      break
```
 before this line as you don't really do any iteration on the column

File: src/repli1d/nn.py [hunk 2]
```diff
@@ -1,18 +1,62 @@
...
+    max_element = []
+    min_element = []
+    transformed = []
+    if isinstance(signal, pd.DataFrame):
+        signal = signal.to_numpy()
+    elif isinstance(signal, list):
+        signal = np.array(signal)
+    if signal.ndim == 1:
+        max_element = max(signal)
+        min_element = min(signal)
```
Comment (organic-chemistry) [inline src/repli1d/nn.py]: Usually min and max are not really robust in biology because you can have very large outsider point.
It is probably better to use 10% quantile and 90 %, except maybe for the initiation as it should be between 0 and one.

**Review by organic-chemistry — COMMENTED** [review]
I did the review in the code.
From what I read, by default the code that I used before should be untouched.
Is it right?

File: src/repli1d/nn.py (hunk 1 above)
Comment (amir-zeraati) [reply src/repli1d/nn.py]: Exactly, thanks.

File: src/repli1d/nn.py (hunk 2 above)
Comment (amir-zeraati) [reply src/repli1d/nn.py]: if outliers are due to experimental error or noise, and they have no information about the output, it's correct. otherwise using min/max can be a better idea because it can cover more range of inputs(less need to extrapolation for NN).
In any case having the constants for transforming new data on the same scale as training data is a good idea, I can apply the normalization method based on your suggestion, please let me know if the outliers are noise.

Comment (organic-chemistry) [reply src/repli1d/nn.py]: Outliers are typically experimental error, because it comes from sequencing data et as some sequence that are localised at different positon are mapped sometimes at a unique position, and this can create artificially very high peaks.
But I agree with you that it is better pratice to store this information.
